search_test.go

```diff
@@ -1017,6 +1017,57 @@ var _ = Describe("RediSearch commands", Label("search"), func() {
 		Expect(res.Attributes[0].WithSuffixtrie).To(BeTrue())
 	})
 
+	It("should test dialect 4", Label("search", "ftcreate", "ftsearch"), func() {
```

Comment:
Make sure to test exact matching for email format strings and for numbers

https://github.com/predis/predis/pull/1466/files#diff-0fe9676f0ea9524cf8f2f20d2640b9a910370320667acc1d5eb6f114fd96b1e1R171

https://github.com/predis/predis/pull/1466/files#diff-0fe9676f0ea9524cf8f2f20d2640b9a910370320667acc1d5eb6f114fd96b1e1R180